byway/src/main/webapp/routepage.js

```diff
@@ -72,39 +79,36 @@ function initMap() {
   
 }
 
-/** Displays route containing waypoints overtop the map. */
-function calcRouteWithRecs() {
-  let request = {
-    origin:  start.name,
-    destination: end.name,
-    travelMode: 'DRIVING',
-    waypoints:  route.map(waypoint => ({location: waypoint.geometry.location})),
-    optimizeWaypoints: true
-  };
-  directionsService.route(request, function(response, status) {
-    if (status == 'OK') {
-      directionsRenderer.setDirections(response);
-      orderWaypoints(response);
-      updateDistanceTime(response);
-    } else {
-      window.alert("Could not calculate route due to: " + status);
-    }
+/** Uses the route from the directionsService request to update 
+ *  ordering of stops on the route panel, and distance and time of roadtrip
+ */
+async function updatePageInfo(){
+  try{
+    let result = await computeRouteForTrip(directionsService, start.place_id, end.place_id, 
+      route.map(waypoint => ({location: waypoint.geometry.location})));
+      directionsRenderer.setDirections(result);
```

Comment:
Nit: unindent this line to match `let result = ...`; since it's indented the same as the continuation of the above line, it looks like its also a continuation of the previous command rather than a new line.

Reply:
Did not catch that indent, fixed!